app/portainer/components/endpoint-list/endpoint-item/endpoint-item-controller.js

```diff
@@ -26,15 +26,34 @@ class EndpointItemController {
     return _.join(tagNames, ',');
   }
 
+  isEdgeEndpoint() {
+    return this.model.Type === 4 || this.model.Type === 7;
+  }
+
+  calcIsCheckInValid() {
+    if (!this.isEdgeEndpoint()) {
+      return false;
+    }
+    const checkInInterval = this.model.EdgeCheckinInterval;
+    const now = Math.floor(Date.now() / 1000);
+
+    return now - this.model.LastCheckInDate <= checkInInterval * 2;
```

Comment:
why is `checkInInterval * 2` used here, should it be `< checkInInterval` @chiptus ?

Reply:
because of this:
https://github.com/portainer/portainer/pull/4782#pullrequestreview-578738682

Reply:
```suggestion
    // give some room to heartbeat to avoid switching from red to green when refreshing the view every few seconds by having current time - 2*checking frequency before triggering the update
    return now - this.model.LastCheckInDate <= checkInInterval * 2;
```